### PR TITLE
Removing ner duckling

### PR DESCRIPTION
**Pull request by EPedrotti**
**Proposed changes**:
- Remove the `ner_duckling` entity extractor. 
**ner_duckling_http is staying, don't worry**

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

**Comment by howl-anderson** [comment]
Why we are removing duckling NER? :)

**Comment by akelad** [comment]
@howl-anderson `ner_duckling_http` is staying 🙂 only `ner_duckling` is being removed because the python wrapper and what not is no longer supported

**Comment by howl-anderson** [comment]
@akelad Thank you! I get it. Nice work BTW!

File: src/pip-delete-this-directory.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+This file is placed here by pip to indicate the source was put
```
Comment (ricwo) [inline src/pip-delete-this-directory.txt]: please remove :)

File: tests/base/test_extractors.py [hunk 2]
```diff
@@ -121,18 +122,48 @@ def test_crf_json_from_non_BILOU(spacy_nlp, ner_crf_pos_feature_config):
 
 
 def test_duckling_entity_extractor(component_builder):
-    _config = RasaNLUModelConfig({"pipeline": [{"name": "ner_duckling"}]})
-    _config.set_component_attr("ner_duckling", dimensions=["time"])
-    duckling = component_builder.create_component("ner_duckling", _config)
+    httpretty.register_uri(
+        httpretty.POST,
+        "http://localhost:8000/parse",
+        body='[{"body":"Today","start":0,"value":{"values":[{"value":"2018-11-13T00:00:00.000-08:00","grain":"day",'
```
Comment (ricwo) [inline tests/base/test_extractors.py]: 80-char width

File: tests/base/test_extractors.py [hunk 3]
```diff
@@ -121,18 +122,48 @@ def test_crf_json_from_non_BILOU(spacy_nlp, ner_crf_pos_feature_config):
...
 
+    httpretty.register_uri(
+        httpretty.POST,
+        "http://localhost:8000/parse",
+        body='[{"body":"tomorrow","start":12,"value":{"values":[{"value":"2013-10-13T00:00:00.000Z","grain":"day",'
+             '"type":"value"}],"value":"2013-10-13T00:00:00.000Z","grain":"day","type":"value"},"end":20,'
+             '"dim":"time","latent":false}] '
+    )
+
     # 1381536182000 == 2013/10/12 02:03:02
```
Comment (ricwo) [inline tests/base/test_extractors.py]: please update this comment to the new timestamp

File: tests/base/test_extractors.py [hunk 4]
```diff
@@ -121,18 +122,48 @@ def test_crf_json_from_non_BILOU(spacy_nlp, ner_crf_pos_feature_config):
...
     entities = message.get("entities")
-    assert len(entities) == 3
+    assert len(entities) == 4
 
     # Test duckling with a defined date
 
+    httpretty.register_uri(
+        httpretty.POST,
+        "http://localhost:8000/parse",
+        body='[{"body":"tomorrow","start":12,"value":{"values":[{"value":"2013-10-13T00:00:00.000Z","grain":"day",'
```
Comment (ricwo) [inline tests/base/test_extractors.py]: linewidth

File: test_result.txt [hunk 5]
```diff
@@ -0,0 +1,25 @@
+. tests/base/test_extractors.py::test_crf_extractor
```
Comment (ricwo) [inline test_result.txt]: please remove this file

File: test_models/test_model_spacy_sklearn/model_20181030-144819/entity_synonyms.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (ricwo) [inline test_models/test_model_spacy_sklearn/model_20181030-144819/entity_synonyms.json]: please update all references to the `test_models` dir in the code (mostly in the tests I believe)

**Comment by tmbo** [comment]
@EPedrotti what is the status here?

File: test_models/test_model_spacy_sklearn/model_20181030-144819/entity_synonyms.json (hunk 6 above)
Comment (EPedrotti) [reply test_models/test_model_spacy_sklearn/model_20181030-144819/entity_synonyms.json]: what do you mean with this one?

**Comment by tmbo** [comment]
The tests failing, do you mind taking a look at them?

**Comment by EPedrotti** [comment]
@tmbo it's on the todolist for today

File: test_models/test_model_spacy_sklearn/model_20181030-144819/entity_synonyms.json (hunk 6 above)
Comment (ricwo) [reply test_models/test_model_spacy_sklearn/model_20181030-144819/entity_synonyms.json]: I mean the explicit paths in `test_utils.py`

File: .gitignore [hunk 7]
```diff
@@ -42,3 +42,5 @@ docs/key
 docs/key.pub
 secrets.tar
 .pytest_cache
+src
```
[the remark this answers is not in the document]
Comment (EPedrotti) [reply .gitignore]: @ricwo to avoid the upload of the sphinx folder inside

[the remark this answers is not in the document]
Comment (EPedrotti) [reply .gitignore]: @ricwo about the models: I did not retrain the models, I was using the one that was already there

[the remark this answers is not in the document]
Comment (ricwo) [reply .gitignore]: how come the binaries changed if you didn't retrain?

**Comment by EPedrotti** [comment]
> Looks great, I just have one question: Why are the two model binaries empty (`test_models/test_model_spacy_sklearn/model_20170628-002705/crf_model.pkl` and `test_models/test_model_spacy_sklearn/model_20170628-002705/intent_classifier.pkl`)? Did you mean to remove them?

I wasn't aware of this. I don't know why it happen. I restored them in the last commit

**Pull request merged**